website/plugins/gatsby-plugin-svg-sprite/gatsby-node.js

```diff
@@ -1,42 +1,31 @@
 exports.onCreateWebpackConfig = (
-  { actions, loaders, getConfig },
+  { actions, getConfig, rules },
   options = {}
 ) => {
-  const prevConfig = getConfig();
+  const config = getConfig();
+  const imagesRule = rules.images();
+  const imagesRuleTest = String(imagesRule.test);
 
   const { rule, ...rest } = options;
 
-  actions.replaceWebpackConfig({
-    ...prevConfig,
-    module: {
-      ...prevConfig.module,
-      rules: [
-        ...prevConfig.module.rules.map((item) => {
-          const { test } = item;
-
-          if (
-            test &&
-            test.toString() === "/\\.(ico|svg|jpg|jpeg|png|gif|webp)(\\?.*)?$/"
-          ) {
-            return {
-              ...item,
-              test: /\.(ico|jpg|jpeg|png|gif|webp)(\?.*)?$/,
-            };
-          }
-
-          return { ...item };
-        }),
+  config.module.rules = [
```

Comment:
are we sure that this has no side-effects?

Reply:
> are we sure that this has no side-effects?

this is the same done in all these kind of plugins, for example see: [gatsby-plugin-react-svg](https://www.gatsbyjs.com/plugins/gatsby-plugin-react-svg/#gatsby-plugin-react-svg-npm-version)

https://github.com/jacobmischka/gatsby-plugin-react-svg/blob/master/gatsby-node.js